graphjoiner/declarative/__init__.py

```diff
@@ -412,8 +413,26 @@ def __new__(meta, name, bases, attrs):
 
         field_definitions, fields = _declare_fields(cls)
         fields = lazy(fields)
-        cls.__fields__ = staticmethod(lambda: list(field_definitions))
 
+        attrs_cls_attrs = dict(
+            raw_=attr.attrib(default=undefined),
+            **dict(
+                (field.attr_name, attr.attrib(default=getattr(field, "default", undefined)))
+                for field in fields().values()
```

Comment:
The fields a type may refer to a type that hasn't been defined yet, which is why `fields` is defined using `lazy`. In other words, you can't evaluate `fields` within `__new__`.